code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -641,6 +651,81 @@
 	else
 		return
 
+/datum/action/innate/slime_puddle
+	name = "Puddle Transformation"
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "slimepuddle"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	background_icon_state = "bg_alien"
+	var/is_puddle = FALSE
+	var/in_transformation_duration = 12
+	var/out_transformation_duration = 7
+	var/puddle_into_effect = /obj/effect/temp_visual/slime_puddle
+	var/puddle_from_effect = /obj/effect/temp_visual/slime_puddle/reverse
+	var/puddle_icon = 'icons/mob/mob.dmi'
+	var/puddle_state = "puddle"
+	var/tracked_overlay
+	var/datum/component/squeak/squeak
+	var/transforming = FALSE
+
+/datum/action/innate/slime_puddle/IsAvailable()
+	if(!transforming)
+		return ..()
+	else
+		return FALSE
+
+/datum/action/innate/slime_puddle/Activate()
+	if(isjellyperson(owner) && IsAvailable())
+		transforming = TRUE
+		UpdateButtonIcon()
+		var/mob/living/carbon/human/H = owner
+		var/mutcolor = "#" + H.dna.features["mcolor"]
+		if(!is_puddle)
+			is_puddle = TRUE
+			owner.cut_overlays()
+			var/obj/effect/puddle_effect = new puddle_into_effect(get_turf(owner), owner.dir)
+			puddle_effect.color = mutcolor
+			H.Stun(in_transformation_duration, ignore_canstun = TRUE)
+			H.rotate_on_lying = FALSE
+			H.KnockToFloor(TRUE, TRUE, TRUE) //disarms and you cant pickup after
+			ADD_TRAIT(H, TRAIT_PARALYSIS_L_ARM, "SLIME_PUDDLE_PARALYSIS_L_ARM")
+			ADD_TRAIT(H, TRAIT_PARALYSIS_R_ARM, "SLIME_PUDDLE_PARALYSIS_R_ARM")
+			ADD_TRAIT(H, TRAIT_MOBILITY_NOPICKUP, "SLIME_PUDDLE_NO_PICKUP")
+			ADD_TRAIT(H, TRAIT_MOBILITY_NOUSE, "SLIME_PUDDLE_NO_ITEMUSE")
```

Comment:
please use proper trait source in a #define

Reply:
aight